### PR TITLE
Type API methods and properties

### PR DESCRIPTION
**Pull request by fbouynot**
Type API methods and properties

DO NOT DELETE THE UNDERLYING TEXT

#### Please note

> Please read this information carefully. You can run `./lnms dev:check` to check your code before submitting.

- [x] Have you followed our [code guidelines?](https://docs.librenms.org/Developing/Code-Guidelines/)
- [x] If my Pull Request does some changes/fixes/enhancements in the WebUI, I have inserted a screenshot of it.
- [x] If my Pull Request makes discovery/polling/yaml changes, I have added/updated [test data](https://docs.librenms.org/Developing/os/Test-Units/).

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`
After you are done testing, you can remove the changes with `./scripts/github-remove`.  If there are schema changes, you can ask on discord how to revert.

### CODE REVIEW

File: app/ApiClients/BaseApi.php [hunk 1]
```diff
@@ -30,10 +30,9 @@
 
 class BaseApi
 {
-    protected $base_uri;
-    /** @var int */
-    protected $timeout = 3;
-    private $client;
+    protected ?string $base_uri;
```
Comment (Jellyfrog) [inline app/ApiClients/BaseApi.php:33]: this should not be nullable

File: app/ApiClients/BingApi.php [hunk 2]
```diff
@@ -34,7 +34,7 @@ class BingApi extends BaseApi implements Geocoder
 {
     use GeocodingHelper;
 
-    protected $base_uri = 'http://dev.virtualearth.net';
+    protected ?string $base_uri = 'http://dev.virtualearth.net';
```
Comment (Jellyfrog) [inline app/ApiClients/BingApi.php:37]: this should not be nullable

File: app/ApiClients/BingApi.php [hunk 3]
```diff
@@ -34,7 +34,7 @@ class BingApi extends BaseApi implements Geocoder
...
     protected $geocoding_uri = '/REST/v1/Locations';
```
Comment (Jellyfrog) [inline app/ApiClients/BingApi.php:38]: While at it, could you typehint all of the extra properties?

File: app/ApiClients/BaseApi.php (hunk 1 above)
Comment (fbouynot) [reply app/ApiClients/BaseApi.php:33]: OK, I think I had error with it non-nullable, so I'll deal with them

File: app/ApiClients/BingApi.php (hunk 3 above)
Comment (fbouynot) [reply app/ApiClients/BingApi.php:38]: Sure, I was trying to achieve something simple first. Now it works, I'll do the same with the extra properties.

File: app/ApiClients/BaseApi.php (hunk 1 above)
Comment (fbouynot) [reply app/ApiClients/BaseApi.php]: I don't understand how it can be related: https://github.com/librenms/librenms/actions/runs/3273454425/jobs/5385675803

Comment (Jellyfrog) [reply app/ApiClients/BaseApi.php]: Because oxidized.php line 37 sets it to null

**Comment by Jellyfrog** [comment]
> What do I do with the getconfig error @Jellyfrog ? Do I add an exception in the neon file, or should I use something else, or should I use baseapi instead of guzzle?

Baseapi would be good, it probably happens because you typehinted guzzle for it, so it could check deeper into it.

File: app/ApiClients/GraylogApi.php [hunk 4]
```diff
@@ -158,7 +146,7 @@ function ($address) {
         return $addresses->filter()->unique();
     }
 
-    public function isConfigured()
+    public function isConfigured(): bool
     {
         return isset($this->client->getConfig()['base_uri']);
```
Comment (murrant) [inline app/ApiClients/GraylogApi.php:151]:
```suggestion
        return (bool) Config::get('graylog.server');
```

Prevent future bug

**Comment by murrant** [comment]
Odd usage, we don't even need to call getConfig, I suggested a fix.

File: app/ApiClients/GraylogApi.php [hunk 5]
```diff
@@ -120,7 +120,7 @@ public function buildSimpleQuery(?string $search = null, ?Device $device = null)
         return implode(' && ', $query);
     }
 
-    public function getAddresses(Device $device): array
+    public function getAddresses(Device $device): Illuminate\Support\Collection
```
Comment (murrant) [inline app/ApiClients/GraylogApi.php]: This is missing the leading \ like `\Illuminate\Support\Collection`  You can only omit the leading slash when the file is in the global namespace.

**Review by murrant — APPROVED** [review]

**Pull request merged**